vars/vars_tracker.go

```diff
@@ -13,35 +14,64 @@ type CredVarsTrackerIterator interface {
 	YieldCred(string, string)
 }
 
+//go:generate counterfeiter . CredVarsTracker
+
 type CredVarsTracker interface {
 	Variables
 	IterateInterpolatedCreds(iter CredVarsTrackerIterator)
 	Enabled() bool
+
+	AddLocalVar(string, interface{}, bool)
 }
 
 func NewCredVarsTracker(credVars Variables, on bool) CredVarsTracker {
-	if on {
-		return &credVarsTracker{
-			credVars:          credVars,
-			interpolatedCreds: map[string]string{},
-			lock:              sync.RWMutex{},
-		}
-	} else {
-		return dummyCredVarsTracker{credVars: credVars}
+	return &credVarsTracker{
+		localVars:         StaticVariables{},
+		credVars:          credVars,
+		enabled:           on,
+		interpolatedCreds: map[string]string{},
+		noRedactVarNames:  map[string]bool{},
+		lock:              sync.RWMutex{},
 	}
 }
 
 type credVarsTracker struct {
-	credVars          Variables
+	credVars  Variables
+	localVars StaticVariables
+
+	enabled bool
+
 	interpolatedCreds map[string]string
 
+	noRedactVarNames map[string]bool
+
 	// Considering in-parallel steps, a lock is need.
 	lock sync.RWMutex
 }
 
 func (t *credVarsTracker) Get(varDef VariableDefinition) (interface{}, bool, error) {
-	val, found, err := t.credVars.Get(varDef)
-	if found {
+	var (
+		val   interface{}
+		found bool
+		err   error
+	)
+
+	redact := true
+	parts := strings.Split(varDef.Name, ":")
+	if len(parts) == 2 && parts[0] == "." {
+		varDef.Name = parts[1]
+		val, found, err = t.localVars.Get(varDef)
+		if found {
+			parts = strings.Split(varDef.Name, ".")
+			if _, ok := t.noRedactVarNames[parts[0]]; ok {
+				redact = false
```

Comment:
Ah, I guess this part would be kind of challenging, huh... :thinking:

Reply:
If later on we add `reveal` to `var_source`, then we can directly add local vars to `NamedVariables`.